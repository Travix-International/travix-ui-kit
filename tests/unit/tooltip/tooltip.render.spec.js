import React from 'react';
import { shallow, mount } from 'enzyme';
import { shallowToJson, mountToJson } from 'enzyme-to-json';
import Tooltip from '../../../components/tooltip/tooltip';

describe('Tooltip: render', () => {
  it('should render normal inactive tooltip without config', () => {
    const wrapper = shallow(<Tooltip/>);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('should render normal active tooltip with mods', () => {
    const wrapper = shallow(<Tooltip active mods={['custom-class']} />);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('should render active tooltip with all passed props', () => {
    const props = {
      active: true,
      children: <div> Tooltip content </div>,
      height: '70px',
      margin: '10px',
      oppositeAxisOffset: '5px',
      position: 'right',
      showCloseButton: true,
      triggerAction: 'hover',
      width: '120px',
    };

    const wrapper = mount(<Tooltip {...props} />);

    expect(wrapper.find('Tooltip').props().active).toEqual(props.active);
    expect(wrapper.find('Tooltip').props().children).toEqual(props.children);
    expect(wrapper.find('Tooltip').props().height).toEqual(props.height);
    expect(wrapper.find('Tooltip').props().margin).toEqual(props.margin);
    expect(wrapper.find('Tooltip').props().oppositeAxisOffset)
      .toEqual(props.oppositeAxisOffset);
    expect(wrapper.find('Tooltip').props().showCloseButton)
      .toEqual(props.showCloseButton);
    expect(wrapper.find('Tooltip').props().triggerAction).toEqual(props.triggerAction);
    expect(wrapper.find('Tooltip').props().width).toEqual(props.width);
    expect(mountToJson(wrapper)).toMatchSnapshot();
  });

  it('should properly initialize style', () => {
    const props = {
      active: true,
      children: <div> Tooltip content </div>,
      height: '70px',
      margin: '10px',
      oppositeAxisOffset: '5px',
      position: 'right',
      showCloseButton: true,
      triggerAction: 'hover',
      width: '120px',
    };

    const prototype = {
      props,
      container: {},
      countPositionOffset: jest.fn()
        .mockReturnValue({
          top: { top: -80, left: '5px' },
          bottom: { bottom: -80, left: '5px' },
          right: { right: -130, bottom: '5px' },
          left: { left: -130, bottom: '5px' },
        }),
      renderCloseButtonBlock: jest.fn()
        .mockReturnValue(<div />),
    };

    const result = Tooltip.prototype.render.call(prototype);
    const expected = { width: '120px', height: '70px', right: -130, bottom: '5px' };
    expect(result.props.style).toEqual(expected);
  });
});
