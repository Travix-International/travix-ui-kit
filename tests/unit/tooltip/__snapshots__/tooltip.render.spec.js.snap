// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip: render should render active tooltip with all passed props 1`] = `
<Tooltip
  active={true}
  height="70px"
  margin="10px"
  oppositeAxisOffset="5px"
  position="right"
  showCloseButton={true}
  triggerAction="hover"
  width="120px"
>
  <div
    className="ui-tooltip ui-tooltip_active ui-tooltip_right"
    style={
      Object {
        "height": "70px",
        "width": "120px",
      }
    }
  >
    <div>
       Tooltip content 
    </div>
  </div>
</Tooltip>
`;

exports[`Tooltip: render should render normal active tooltip with mods 1`] = `
<div
  className="ui-tooltip ui-tooltip_custom-class ui-tooltip_active ui-tooltip_top"
  style={
    Object {
      "height": "",
      "width": "",
    }
  }
/>
`;

exports[`Tooltip: render should render normal inactive tooltip without config 1`] = `
<div
  className="ui-tooltip ui-tooltip_inactive ui-tooltip_top"
  style={
    Object {
      "height": "",
      "width": "",
    }
  }
/>
`;
